stl/inc/span

```diff
@@ -0,0 +1,723 @@
+        // non-member overload to model the exposition-only forwarding-range concept
+        return _Right.begin();
+    }
+    _NODISCARD friend constexpr const_iterator end(span _Right) noexcept {
+        // non-member overload to model the exposition-only forwarding-range concept
+        return _Right.end();
+    }
+#endif // __cpp_lib_concepts
+
+    _NODISCARD constexpr const_iterator cbegin() const noexcept {
+        return begin();
+    }
+
+    _NODISCARD constexpr const_iterator cend() const noexcept {
+        return end();
+    }
+
+    _NODISCARD constexpr reverse_iterator rbegin() noexcept {
+        return reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr reverse_iterator rend() noexcept {
+        return reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rbegin() const noexcept {
+        return const_reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rend() const noexcept {
+        return const_reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crbegin() const noexcept {
+        return rbegin();
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crend() const noexcept {
+        return rend();
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_begin() noexcept {
+        return data();
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_begin() const noexcept {
+        return data();
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_end() noexcept {
+        return data() + size();
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_end() const noexcept {
+        return data() + size();
+    }
+
+    constexpr void swap(span& _Other) noexcept {
+        _Mydata.swap(_Other._Mydata);
+    }
+
+private:
+    _Span_storage<pointer, _Extent> _Mydata;
+};
+
+// [span.tuple] Tuple interface
+template <class _Ty, size_t _Extent>
+struct tuple_size<span<_Ty, _Extent>> : integral_constant<size_t, _Extent> {};
+
+template <class _Ty>
+struct tuple_size<span<_Ty, dynamic_extent>>;
+
+template <size_t _Index, class _Ty, size_t _Extent>
+struct tuple_element<_Index, span<_Ty, _Extent>> {
+    static_assert(dynamic_extent != _Extent, "std::tuple_element<> not supported for std::span<T, dynamic_extent>");
+    static_assert(_Index < _Extent, "Index out of bounds in std::tuple_element<> (std::span)");
+    using type = _Ty;
+};
+
+template <size_t _Index, class _Ty, size_t _Extent>
+_NODISCARD constexpr _Ty& get(span<_Ty, _Extent> _Span) noexcept {
+    static_assert(dynamic_extent != _Extent, "std::get<> not supported for std::span<T, dynamic_extent>");
+    static_assert(_Index < _Extent, "Index out of bounds in std::get<> (std::span)");
+    return _Span[_Index];
+}
+
+// [span.objectrep] Views of object representation
+// FUNCTION TEMPLATE as_bytes
+template <class _Ty, size_t _Extent>
+_NODISCARD auto as_bytes(span<_Ty, _Extent> _Sp) noexcept

```

Comment:
Ditto "Potential LWG issue: `size()` had better not be greater than `size_t(-1) / sizeof(ElementType)`". (And on 686.)

Reply:
Done